sensor/common/clusterentities/store.go

```diff
@@ -240,6 +313,18 @@ func (e *Store) removeHistoricalExpiredIPs(deploymentID string, ip net.IPAddress
 	}
 }
 
+func (e *Store) removeHistoricalExpiredContainerIDs(contID string, meta ContainerMetadata) {
+	if status, ok := e.historicalContainerIDs[contID][meta]; ok && status.IsExpired() {
+		// remove in the non-historical storage
+		delete(e.containerIDMap, contID)
```

Comment:
I think this could be a potential bug. If memory == 0, the `historicalContainerIDs` will be empty (because `markContainerIDHistorical` does not do anything if memory == 0) and we will never delete the entry from `containerIDMap`. I think this applies to endpoints and ips too.

Reply:
I think this is a bit messy, but not a bug.

Messy because:
- this func should only touch historical maps, not the regular ones

Not a bug because:
- if history size is 0, then the entry will be deleted in `purgeNoLock` - there the for loop iterates over `reverseContainerIDMap`

Nonetheless - I will double check that this is not an issue for other historical maps and make less confusing by removing the messy part.

Reply:
I also suspect that I implemented the history for IPs and Entities a bit differently than the one for container IDs and that may be a bit bigger mess.

For IPs and Entities, I may keep duplicate entries in the regular maps and the historical maps for as long as the history is valid. 
When working on ContainerIDs, I copied some of the code but then changed the behavior to: a container id is either historical or not - there is no situation that a given container ID will be in the both maps at the same time.

I find the second way much better, cleaner, and easier to test.

I have opened a follow-up to bring them up on the same page. 
When reviewing this PR, you may need to keep in mind that there are bugs for the historical entities and IPs - I know about them and will fix them asap.

Reply:
> if history size is 0, then the entry will be deleted in purgeNoLock - there the for loop iterates over reverseContainerIDMap
Ah! Now I see that but then I'd say deleting from the `containerIDMap` in `removeHistoricalExpiredContainerIDs` is not necessary at all because it's going to be delete regardless in [line 386](https://github.com/stackrox/stackrox/blob/79f22dbaf0232f657149594ee157fa65b7f4facc/sensor/common/clusterentities/store.go#L386).
> I will double check that this is not an issue for other historical maps and make less confusing by removing the messy part.

I've checked and for ips end endpoints the `markHistorical` functions are not guarded with `if entifiesMemorySize == 0` therefore, if I understood correctly, they get added always to the historical maps but with 0 ticks left. So basically they expire in the first tick.

> I have opened a follow-up to bring them up on the same page.
When reviewing this PR, you may need to keep in mind that there are bugs for the historical entities and IPs - I know about them and will fix them asap.

I'm ok with changing this in a follow-up. I'll keep that in mind for the review 👍